### PR TITLE
Adding MM Patch for USI-LS

### PR DESCRIPTION
**Pull request by jmrd98**
Initial try at values/descriptions.

---

So, this was my bored monday evening. I have barely had a chance to test these yet, but I didn't want to delay further. They seem technically valid, but I have no good idea on balance as far as numbers go. I also tried my hand at some creative descriptions, but I will leave it to you all to decide on the desired amusement level :)

I also may have missed some parts, in which case, well... Let me know, and I'll try to fix it up asap.

EDIT: I forgot one already, the MalhanaSM part... But!.. I have no idea how to use this guy/what the purpose is. I guessed in a 14 kerbal-day supply, but i'm not sure if that is even close.

### CODE REVIEW

**Comment by jsolson** [comment]
Let me know when you're done, and I'll merge it :-)

**Comment by Araym-KSP** [comment]
... remember to add Supplies cost to parts (values tend to sky rocket) as it could go in the "negative", if not used, sometime...
"cost += xxxxxxx" (with xxxxxxx as the value for supplies * 2.5 kredits x supplies) do the trick for MM
(I patched privately almost all the crewed parts from stock and any mods with a generic file that count number of crews and do autonomously all the math needed, if there isn't a specific USI definition :P)

**Comment by Cowboy-WY** [comment]
That is a very interesting point.  That means the cost would need adjusting every time the amount is modified.  Might be simpler to set amount = 0 and only adjust the max amount to avoid cost calculations.

**Comment by Araym-KSP** [comment]
Remember that KSP consider the "cost" value as "wet cost":
if you add a maxAmount for supplies (or any resource), then set the amount to "zero" (as they start unfueled for the considered resource), the game still remove their cost from the part one, leading, in some cases, to "negative values" building a craft.

That why I had to figure, for my USI-LF patch, those 3 lines of code that change dinamically the cost of the part patched (before it, I was gaining kredits, launching parts without supplies, rather than spend them :P)

**Comment by Cowboy-WY** [comment]
I didn't know that.  Sounds like there is no way to avoid adjusting the cost based on the max amount.

**Comment by Araym-KSP** [comment]
Nope: there isn't any way. "Cost" in cfg are the FULL, "wet" cost of "maxAmount" resources, even if then, in "amount", are left empty at "zero" (KSP subtract them from the cost, if zeroed)
At least untill Squad consider (better for any modder) to change the way KSP calculates each part cost from "wet" to "dry" (allowing to add resources without bothering of "negative cost").

The problem with USI-LF supplies is the very hi cost of them, that could totally break a craft cost (as I said, I gained kredits untill properly patched)

**Comment by jmrd98** [comment]
Excellent point... I was seeing that as well, but ran into the "****, it's 2am" barrier before I could look at it in detail. The patch as you say is simple, and simpler yet with your hint. (Many thanks!) I should have it in this evening (~8-9 hr), as I'm at lunch from work doing other coding at the moment :p

**Comment by jmrd98** [comment]
Alright, check it out now. The syntax on this was flummoxing me...gerdarn @ symbols... ><

It is fairly bulletproof, in that it will auto adjust to changes in the supply cost/unit, if ever they occur. (I hate hardcoding things. This shows up in work and in play, as evidenced here... heh.)

We multiply our maxAmount of supplies by 2.5 (currently), and add that value to the part cost. This is flexible enough to contend with w/e supply values a part is gifted.

Blah blah, please someone check my work, 3:30 am, etc etc etc...

**Comment by jsolson** [comment]
I wanted to install USI to check it out but I'm a bit confused. The only thing it seems to add is Replacement Parts. Nothing about Supplies or Mulch or other stuff in any of it's configs. What probably obvious step am I missing here?

Before I got involved here and played career saves I was a TAC life support guy.

**Comment by jsolson** [comment]
Oh btw. A catchall for BDB parts would be:

@PART[bluedog*,Bluedog*]

Shouldn't you be able to do a lot of this with a (psudocode) @PART[*]:HAS[Crew] and them multiply everything out by the crew capacity? That how we'd do it with TAC. I'm unfamiliar with USI so maybe not.

**Comment by jmrd98** [comment]
@jsolson I don't know what is up with the install. Did you get the 000_USITools folder as well? Some common pieces are in there, I think. But I just had a look at the config, and I think I see what might be up: They add some mechanics and modifiers to crewed parts, but the actual supplies are in dedicated parts (In the Utility category), and not the parts themselves, unless patched like this.

Conversely, I haven't played with TAC-LS since 0.25, haha.

---

Good points; I'll adjust the AFTER, as that was a late night fit of desperation, and you are correct.

I can also easily change the catchall, but I only had the lower cased variety as it turned out that the only parts the patch hit were of that variety. As the catchall at the end only really applied to the components above it, it slipped my mind to future proof it. /slaps self on the wrist.

As for not using a crew based metric for supplies, in my late night creativity, I had it in mind that the different parts would have different utility, not only based on the crew compliment. I stretched the Gemini 2 man to a 14 day supply, while the BigGem only got a 4 day supply. Perhaps it's a bit of roleplaying license, but if all the parts were the same, it'd get a bit boring, IMO.

tldr, I probably did a lot of extra work that could have been avoided with a *= #CrewCap, but I thought the flavor was worth it.

Now that the things are in though, the numbers are at least more easily tweaked, as tweaking the aether is somewhat difficult.
